### PR TITLE
Add Jenkinsfile for participation infrastructure

### PR DESCRIPTION
**Pull request by flamingspaz**
(did this on master branch the first time, bad idea)

### CODE REVIEW

**Comment by flamingspaz** [comment]
This is ready for review

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            print "Invalid branch"
+            currentBuild.result = "FAILURE"
+            throw err
+        break
+    }
+    type = 'group'
+}
+
+node('mesos') {
+def image
```
Comment (johngian) [inline Jenkinsfile]: Please indent

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        image = docker.build(app_id + ":" + gitCommit)
+    }
+
+    stage('Push') {
+        sh "docker tag ${image.imageName()} " + dockerRegistry + "/${image.imageName()}"
+        sh "docker push " + dockerRegistry + "/${image.imageName()}"
+    }
+
+    stage('Deploy') {
+        build job: 'deploy-test', parameters: [string(name: 'environment', value: environment), string(name: 'commit_id', value: gitCommit), string(name: 'marathon_id', value: app_id), string(name: 'marathon_config', value: 'remo_' + environment + '.json'), string(name: 'type', value: 'group')]
```
Comment (johngian) [inline Jenkinsfile]: Is it possible to make this more readable by splitting it into more lines?

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+def app_id = "remo"
+def dockerRegistry = "docker-registry.ops.mozilla.community:443"
+    stage('Prep') {
+        checkout scm
+        gitCommit = sh(returnStdout: true, script: 'git rev-parse HEAD').trim()
+    }
+
+    stage('Build') {
+        sh "cp docker/prod Dockerfile"
+        image = docker.build(app_id + ":" + gitCommit)
```
Comment (johngian) [inline Jenkinsfile]: Instead of copying can you just define the Dockerfile ?

**Comment by johngian** [comment]
r+wc

**Pull request merged**